pkg/api/pluginproxy/ds_auth_provider.go

```diff
@@ -46,7 +46,7 @@ func ApplyRoute(ctx context.Context, req *http.Request, proxyPath string, route
 	tokenProvider := newAccessTokenProvider(ds, route)
 
 	if route.TokenAuth != nil {
-		if token, err := tokenProvider.getAccessToken(data); err != nil {
```

Comment:
Following my previous comment, `ctx` does not serve anything so it should be removed.